### DISCUSSION TITLE
How do I add new items to the HTML via custom fields in the YAML header?

### DISCUSSION THREAD

**Discussion by yuxi-liu-wired** [quarto-dev/quarto-cli#7819] [Q&A]
### Description

I want to have YAML header like 

```yaml
status: "finished"
confidence: "highly likely"
importance: 10
```

and I hope to make it so that the final rendered HTML would contain the lines 

```html
  <div>
    <div class="quarto-title-meta-heading">Importance</div>
    <div class="quarto-title-meta-contents">
      <p class="date-modified">10</p>
    </div>
  </div>

    
  <div>
    <div class="quarto-title-meta-heading">Confidence</div>
    <div class="quarto-title-meta-contents">
      <p class="date-modified">highly likely</p>
    </div>
  </div>
    
  <div>
    <div class="quarto-title-meta-heading">Status</div>
    <div class="quarto-title-meta-contents">
      <p class="date-modified">finished</p>
    </div>
  </div>
```

I cannot figure out how to add new HTML elements. The [extension](https://quarto.org/docs/extensions/formats.html) doesn't seem to allow that. It only allows changing the default settings.

**Comment by mcanouil** [comment]
I don't know what you are really trying to do as you HTML example seems really off.

See:
- https://quarto.org/docs/authoring/markdown-basics.html#divs-and-spans
- https://quarto.org/docs/authoring/variables.html#meta

**Comment by ijmitch** [comment]
Just by way of comparison, hoping to clarify, @yuxi-liu-wired I think you're looking for something which I'd do in Mkdocs with the [Macros](https://mkdocs-macros-plugin.readthedocs.io/en/latest/) plugin which allows all of the Jinja2 templating to be used in Markdown with access to page-level frontmatter variables.

Just as a guess on what you'd like to do.

**Comment by cderv** [comment]
Depending on where you need to insert this content based on variable, you may need to provide custom template for the rendering. 

See about template at (https://quarto.org/docs/journals/templates.html#templates) and partials logic that Quarto offers (https://quarto.org/docs/journals/templates.html#html-partials)
